### PR TITLE
MGDAPI-5134 - observability: v4.1.0

### PR DESCRIPTION
**Pull request by KevFan**
# Issue link

https://issues.redhat.com/browse/MGDAPI-5134

# What

* Consume Observability Operator v4.1.0

# Verification steps

* Prepare cluster
```
export LOCAL=false
make cluster/prepare/local
```
* Install 1.31.0 using the following index `quay.io/kevfan/managed-api-service-index:1.31.0` to the `redhat-rhoam-operator` namespace
* Deploy RHMI CR
```
make deploy/integreatly-rhmi-cr.yml
```
* Once installation succeeds, upgrade to this branch
```
 oc patch catalogsource rhmi-operators -n openshift-marketplace --type=json -p='[{"op": "replace", "path": /spec/image, "value": "quay.io/kevfan/managed-api-service-index:1.32.0"}]'
```
* Verify observability operator is upgraded
```
oc get csv -n redhat-rhoam-observability-operator
```
* Verify oauth proxy is using newer image
```
 oc get deployment grafana-deployment -n redhat-rhoam-observability -o json | jq '.spec' | grep origin
```
* Verify grafana init plugins is using newer image
```
oc get deployment grafana-deployment -n redhat-rhoam-observability -o json | jq '.spec' | grep plugins_init
```
* Verify RHOAM is upgraded successfully to 1.32.0
```
oc get rhmi rhoam -n redhat-rhoam-operator -o json | jq '.status.version'
```

Note: vendor changes to bring in OO v4.1.0 is not done due to dependency issues

### CODE REVIEW

File: bundles/managed-api-service/1.32.0/manifests/managed-api-service.clusterserviceversion.yaml [hunk 1]
```diff
@@ -217,6 +221,12 @@ spec:
               verbs:
                 - create
                 - list
+            - apiGroups:
+                - ""
+              resources:
+                - pods/exec
+              verbs:
+                - create
```
Comment (KevFan) [inline bundles/managed-api-service/1.32.0/manifests/managed-api-service.clusterserviceversion.yaml:229]: Missing permission in bundle csv from https://github.com/integr8ly/integreatly-operator/pull/3060

**Review by valerymo — COMMENTED** [review]
it looks good to me.
Test / Upgrade worked as expected. Logs - attached
[review_MGDAPI-5134.txt](https://github.com/integr8ly/integreatly-operator/files/10697587/review_MGDAPI-5134.txt)

**Comment by valerymo** [comment]
retested locally: LOCAL=false TEST="(E09|C04|E02|E03|Verify prometheus metrics scrapped)" make test/e2e/single
5 Passed | 0 Failed | 0 Pending | 52 Skipped
log - attached
[test_typescript.txt](https://github.com/integr8ly/integreatly-operator/files/10697927/test_typescript.txt)

**Pull request merged**